diff --git a/src/modules/dbus/dbusmodule.cpp b/src/modules/dbus/dbusmodule.cpp
index e308cf8..906f1b0 100644
--- a/src/modules/dbus/dbusmodule.cpp
+++ b/src/modules/dbus/dbusmodule.cpp
@@ -553,7 +553,7 @@ public:
             return;
         }
 #else
-        FCITX_UNUSED(name);
+        FCITX_UNUSED(fd);
 #endif
         throw dbus::MethodCallError("org.freedesktop.DBus.Error.InvalidArgs",
                                     "Wayland addon is not available.");
