diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index bd5cbde..88740b6 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -1,5 +1,16 @@
 install(DIRECTORY 48x48 scalable DESTINATION "${CMAKE_INSTALL_DATADIR}/icons/hicolor"
+        FILES_MATCHING PATTERN org.fcitx.Fcitx5.*
         PATTERN .* EXCLUDE
         PATTERN *~ EXCLUDE)
 
+foreach (fname rime rime-deploy rime-disable rime-im rime-latin rime-sync)
+  install(FILES 48x48/apps/fcitx-${fname}.png
+          DESTINATION "${CMAKE_INSTALL_DATADIR}/icons/hicolor/48x48/apps"
+          RENAME fcitx5-${fname}.png)
+
+  install(FILES scalable/apps/fcitx-${fname}.svg
+          DESTINATION "${CMAKE_INSTALL_DATADIR}/icons/hicolor/scalable/apps"
+          RENAME fcitx5-${fname}.svg)
+endforeach (fname rime rime-deploy rime-disable rime-im rime-latin rime-sync)
+
 install(FILES fcitx5.yaml DESTINATION "${RIME_DATA_DIR}")
diff --git a/src/rime.conf.in b/src/rime.conf.in
index 604cb06..7ceb3b5 100644
--- a/src/rime.conf.in
+++ b/src/rime.conf.in
@@ -1,6 +1,6 @@
 [InputMethod]
 Name=Rime
-Icon=fcitx-rime
+Icon=fcitx5-rime
 Label=ㄓ
 LangCode=zh
 Addon=rime
diff --git a/src/rimeengine.cpp b/src/rimeengine.cpp
index 45c20e5..bf1fa1f 100644
--- a/src/rimeengine.cpp
+++ b/src/rimeengine.cpp
@@ -96,9 +96,9 @@ public:
             });
         }
         if (isDisabled) {
-            return "fcitx-rime-disabled";
+            return "fcitx5-rime-disabled";
         }
-        return "fcitx-rime-im";
+        return "fcitx5-rime-im";
     }
 
 private:
@@ -113,7 +113,7 @@ RimeEngine::RimeEngine(Instance *instance)
                                                      imAction_.get());
     imAction_->setMenu(&schemaMenu_);
     eventDispatcher_.attach(&instance_->eventLoop());
-    deployAction_.setIcon("fcitx-rime-deploy");
+    deployAction_.setIcon("fcitx5-rime-deploy");
     deployAction_.setShortText(_("Deploy"));
     deployAction_.connect<SimpleAction::Activated>([this](InputContext *ic) {
         deploy();
@@ -125,7 +125,7 @@ RimeEngine::RimeEngine(Instance *instance)
     instance_->userInterfaceManager().registerAction("fcitx-rime-deploy",
                                                      &deployAction_);
 
-    syncAction_.setIcon("fcitx-rime-sync");
+    syncAction_.setIcon("fcitx5-rime-sync");
     syncAction_.setShortText(_("Synchronize"));
 
     syncAction_.connect<SimpleAction::Activated>([this](InputContext *ic) {
@@ -348,7 +348,7 @@ void RimeEngine::notify(const std::string &messageType,
     const char *tipId = "";
     if (messageType == "deploy") {
         tipId = "fcitx-rime-deploy";
-        icon = "fcitx-rime-deploy";
+        icon = "fcitx5-rime-deploy";
         if (messageValue == "start") {
             message = _("Rime is under maintenance. It may take a few "
                         "seconds. Please wait until it is finished...");
@@ -360,7 +360,7 @@ void RimeEngine::notify(const std::string &messageType,
                         "See /tmp/rime.fcitx.ERROR for details.");
         }
     } else if (messageType == "option") {
-        icon = "fcitx-rime";
+        icon = "fcitx5-rime";
         if (messageValue == "!full_shape") {
             tipId = "fcitx-rime-full-shape";
             message = _("Half Shape is enabled.");
@@ -422,7 +422,7 @@ std::string RimeEngine::subModeLabelImpl(const InputMethodEntry &,
 
 std::string RimeEngine::subModeIconImpl(const InputMethodEntry &,
                                         InputContext &ic) {
-    std::string result = "fcitx-rime";
+    std::string result = "fcitx5-rime";
     if (!api_ || !factory_.registered()) {
         return result;
     }
@@ -430,11 +430,11 @@ std::string RimeEngine::subModeIconImpl(const InputMethodEntry &,
     if (state) {
         state->getStatus([&result](const RimeStatus &status) {
             if (status.is_disabled) {
-                result = "fcitx-rime-disable";
+                result = "fcitx5-rime-disable";
             } else if (status.is_ascii_mode) {
-                result = "fcitx-rime-latin";
+                result = "fcitx5-rime-latin";
             } else {
-                result = "fcitx-rime";
+                result = "fcitx5-rime";
             }
         });
     }
