diff --git a/app/dns/nameserver_quic.go b/app/dns/nameserver_quic.go
index bd81153..1b82e27 100644
--- a/app/dns/nameserver_quic.go
+++ b/app/dns/nameserver_quic.go
@@ -9,7 +9,7 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 	"golang.org/x/net/dns/dnsmessage"
 	"golang.org/x/net/http2"
 
diff --git a/common/protocol/quic/qtls_go119.go b/common/protocol/quic/qtls_go119.go
index 98ff28d..4ab25af 100644
--- a/common/protocol/quic/qtls_go119.go
+++ b/common/protocol/quic/qtls_go119.go
@@ -1,12 +1,11 @@
-//go:build go1.19
-// +build go1.19
+//go:build go1.19 && !go1.20
 
 package quic
 
 import (
 	"crypto/cipher"
 
-	"github.com/marten-seemann/qtls-go1-19"
+	"github.com/quic-go/qtls-go1-19"
 )
 
 type (
diff --git a/common/protocol/quic/qtls_go118.go b/common/protocol/quic/qtls_go120.go
similarity index 73%
rename from common/protocol/quic/qtls_go118.go
rename to common/protocol/quic/qtls_go120.go
index 63546a4..e182c75 100644
--- a/common/protocol/quic/qtls_go118.go
+++ b/common/protocol/quic/qtls_go120.go
@@ -1,12 +1,11 @@
-//go:build go1.18 && !go1.19
-// +build go1.18,!go1.19
+//go:build go1.20
 
 package quic
 
 import (
 	"crypto/cipher"
 
-	"github.com/marten-seemann/qtls-go1-18"
+	"github.com/quic-go/qtls-go1-20"
 )
 
 type (
diff --git a/common/protocol/quic/sniff.go b/common/protocol/quic/sniff.go
index d042130..a4fa7b9 100644
--- a/common/protocol/quic/sniff.go
+++ b/common/protocol/quic/sniff.go
@@ -7,7 +7,7 @@ import (
 	"encoding/binary"
 	"io"
 
-	"github.com/lucas-clemente/quic-go/quicvarint"
+	"github.com/quic-go/quic-go/quicvarint"
 	"golang.org/x/crypto/hkdf"
 
 	"github.com/v2fly/v2ray-core/v5/common"
diff --git a/transport/internet/quic/conn.go b/transport/internet/quic/conn.go
index e24fd50..492c242 100644
--- a/transport/internet/quic/conn.go
+++ b/transport/internet/quic/conn.go
@@ -7,7 +7,7 @@ import (
 	"syscall"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 
 	"github.com/v2fly/v2ray-core/v5/common"
 	"github.com/v2fly/v2ray-core/v5/common/buf"
diff --git a/transport/internet/quic/dialer.go b/transport/internet/quic/dialer.go
index 8a52a01..f3e1429 100644
--- a/transport/internet/quic/dialer.go
+++ b/transport/internet/quic/dialer.go
@@ -5,7 +5,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 
 	"github.com/v2fly/v2ray-core/v5/common"
 	"github.com/v2fly/v2ray-core/v5/common/net"
diff --git a/transport/internet/quic/hub.go b/transport/internet/quic/hub.go
index bdd004e..41bae59 100644
--- a/transport/internet/quic/hub.go
+++ b/transport/internet/quic/hub.go
@@ -4,7 +4,7 @@ import (
 	"context"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 
 	"github.com/v2fly/v2ray-core/v5/common"
 	"github.com/v2fly/v2ray-core/v5/common/net"
