diff --git a/lib/connections/quic_dial.go b/lib/connections/quic_dial.go
index 54eec8c..f828c14 100644
--- a/lib/connections/quic_dial.go
+++ b/lib/connections/quic_dial.go
@@ -17,7 +17,7 @@ import (
 	"net/url"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 
 	"github.com/syncthing/syncthing/lib/config"
 	"github.com/syncthing/syncthing/lib/connections/registry"
diff --git a/lib/connections/quic_listen.go b/lib/connections/quic_listen.go
index 7b8fb6d..74a0672 100644
--- a/lib/connections/quic_listen.go
+++ b/lib/connections/quic_listen.go
@@ -18,7 +18,7 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
 
 	"github.com/syncthing/syncthing/lib/config"
 	"github.com/syncthing/syncthing/lib/connections/registry"
diff --git a/lib/connections/quic_misc.go b/lib/connections/quic_misc.go
index ce5490f..d5f0b0d 100644
--- a/lib/connections/quic_misc.go
+++ b/lib/connections/quic_misc.go
@@ -15,17 +15,16 @@ import (
 	"net/url"
 	"time"
 
-	"github.com/lucas-clemente/quic-go"
+	"github.com/quic-go/quic-go"
+
 	"github.com/syncthing/syncthing/lib/osutil"
 )
 
-var (
-	quicConfig = &quic.Config{
-		ConnectionIDLength: 4,
-		MaxIdleTimeout:     30 * time.Second,
-		KeepAlivePeriod:    15 * time.Second,
-	}
-)
+var quicConfig = &quic.Config{
+	ConnectionIDLength: 4,
+	MaxIdleTimeout:     30 * time.Second,
+	KeepAlivePeriod:    15 * time.Second,
+}
 
 func quicNetwork(uri *url.URL) string {
 	switch uri.Scheme {
